client.go

```diff
@@ -236,13 +243,27 @@ func WithTimeout(t time.Duration) Option {
 }
 
 // WithSSL tells the client to use a SSL/TLS connection
+//
+// Deprecated: use WithSSLPort instead.
 func WithSSL() Option {
 	return func(c *Client) error {
 		c.ssl = true
 		return nil
 	}
 }
 
+// WithSSLPort tells the client to use a SSL/TLS connection.
+// It automatically sets the port to 465.
+//
+// When the SSL connection fails and fallback is set to true,
+// the client will attempt to connect on port 25 using plaintext.
+func WithSSLPort(fallback bool) Option {
```

Comment:
Just a cosmetical "issue". So far we've tried to follow the [Go "best practices" for variable names](https://github.com/golang/go/wiki/CodeReviewComments#variable-names). So for consistency I suggest `f` instead of `fallback`